Provider crashes when one (or more) initialization parameters is null, with OS equal to 'other'.

**Describe the bug**
Provider crashed. I'm using this in a module that handles creating resources as needed. Most of these parameters handle null OK, but this one crashes.

**To Reproduce**
Steps to reproduce the behavior:

1. Create a virtual machine resource, including the initialization block as below:
2. Run Terraform apply.

Please also provide a minimal Terraform configuration that reproduces the issue.

Put this block inside of a VM resource with OS set as other type, also provide datastore, I had it set up something like this, with init disabled:
```hcl
  initialization {
      datastore_id = var.datastore
      user_account {
        username = var.init_enabled ? var.user_name : null
        keys     = var.init_enabled ? var.keys : null
      }
    }
  }
```

Doing something like this instead, which only defines the block if needed, solves it:
```hcl
  dynamic "initialization" {
    for_each = each.value.initialization[*]
    content {
      datastore_id = var.datastore
      user_account {
        username = initialization.value.user
        keys     = [initialization.value.key]
      }
    }
  }
```

**Additional context**
```
module.virtual_machines.proxmox_virtual_environment_vm.virtual_machines["0"]: Creating...
╷
│ Error: Request cancelled
│ 
│ The plugin6.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

Stack trace from the terraform-provider-proxmox_v0.46.1 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 168 [running]:
github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource.vmGetCloudInitConfig(0x19e1140?)
        github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm.go:3185 +0x103b
github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource.vmCreateCustom({0x1c96bd0, 0xc0005ac070}, 0x22cfe40?, {0x1a5aba0?, 0xc0000f8c30?})
        github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm.go:2602 +0x12d3
github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource.vmCreate({0x1c96bd0, 0xc0005ac070}, 0x0?, {0x1a5aba0, 0xc0000f8c30})
        github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm.go:1688 +0x94
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0003c3340, {0x1c96b28, 0xc0000ae390}, 0xd?, {0x1a5aba0, 0xc0000f8c30})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:778 +0x11b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0003c3340, {0x1c96b28, 0xc0000ae390}, 0xc0001ba8f0, 0xc0001cfc00, {0x1a5aba0, 0xc0000f8c30})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:909 +0xa89
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000123ab8, {0x1c96b28?, 0xc0000ae1e0?}, 0xc00049af50)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/grpc_provider.go:1074 +0xdbc
github.com/hashicorp/terraform-plugin-mux/tf5to6server.v5tov6Server.ApplyResourceChange({{0x1c9e340?, 0xc000123ab8?}}, {0x1c96b28, 0xc0000ae1e0}, 0x0?)
        github.com/hashicorp/terraform-plugin-mux@v0.13.0/tf5to6server/tf5to6server.go:37 +0x54
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ApplyResourceChange(0x1c96b60?, {0x1c96b28?, 0xc000195e00?}, 0xc00049af00)
        github.com/hashicorp/terraform-plugin-mux@v0.13.0/tf6muxserver/mux_server_ApplyResourceChange.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0xc00033b9a0, {0x1c96b28?, 0xc0001953e0?}, 0xc0005ad570)
        github.com/hashicorp/terraform-plugin-go@v0.21.0/tfprotov6/tf6server/server.go:845 +0x3d0
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x1adffc0?, 0xc00033b9a0}, {0x1c96b28, 0xc0001953e0}, 0xc0001cf100, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.21.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:518 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000229000, {0x1c96b28, 0xc000195350}, {0x1c9c8e8, 0xc0004751e0}, 0xc00001c000, 0xc0004780f0, 0x22c30b8, 0x0)
        google.golang.org/grpc@v1.61.0/server.go:1385 +0xe03
google.golang.org/grpc.(*Server).handleStream(0xc000229000, {0x1c9c8e8, 0xc0004751e0}, 0xc00001c000)
        google.golang.org/grpc@v1.61.0/server.go:1796 +0xfec
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.61.0/server.go:1029 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 28
        google.golang.org/grpc@v1.61.0/server.go:1040 +0x135

Error: The terraform-provider-proxmox_v0.46.1 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

```

- Single or clustered Proxmox: single
- Provider version (ideally it should be the latest version): latest - 0.46.1
- Terraform version: 1.5.4
- OS (where you run Terraform from): Mac OSX Sonoma M1 Max
- Debug logs (`TF_LOG=DEBUG terraform apply`):
```2024-02-01T20:31:16.328-0500 [INFO]  backend/local: apply calling Apply
2024-02-01T20:31:16.328-0500 [DEBUG] Building and walking apply graph for NormalMode plan
2024-02-01T20:31:16.331-0500 [DEBUG] Resource state not found for node "module.virtual_machines.proxmox_virtual_environment_vm.virtual_machines[\"0\"]", instance module.virtual_machines.proxmox_virtual_environment_vm.virtual_machines["0"]
2024-02-01T20:31:16.331-0500 [DEBUG] adding implicit provider configuration provider["registry.terraform.io/hashicorp/random"], implied first by module.virtual_machines.random_pet.virtual_machine_names (expand)
2024-02-01T20:31:16.331-0500 [DEBUG] ProviderTransformer: "module.virtual_machines.proxmox_virtual_environment_vm.virtual_machines[\"0\"]" (*terraform.NodeApplyableResourceInstance) needs provider["registry.terraform.io/bpg/proxmox"]
2024-02-01T20:31:16.331-0500 [DEBUG] ProviderTransformer: "module.virtual_machines.proxmox_virtual_environment_vm.virtual_machines (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/bpg/proxmox"]
2024-02-01T20:31:16.331-0500 [DEBUG] ProviderTransformer: "module.virtual_machines.random_pet.virtual_machine_names (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/hashicorp/random"]
2024-02-01T20:31:16.331-0500 [DEBUG] ProviderTransformer: "module.download_files.proxmox_virtual_environment_download_file.download_files (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/bpg/proxmox"]
2024-02-01T20:31:16.332-0500 [DEBUG] ReferenceTransformer: "var.public_key" references: []
2024-02-01T20:31:16.332-0500 [DEBUG] ReferenceTransformer: "module.download_files.var.files (expand)" references: []
2024-02-01T20:31:16.332-0500 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/bpg/proxmox\"]" references: []
2024-02-01T20:31:16.332-0500 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/hashicorp/random\"]" references: []
2024-02-01T20:31:16.332-0500 [DEBUG] ReferenceTransformer: "module.download_files (close)" references: []
2024-02-01T20:31:16.332-0500 [DEBUG] ReferenceTransformer: "module.virtual_machines (close)" references: []
2024-02-01T20:31:16.332-0500 [DEBUG] ReferenceTransformer: "module.virtual_machines.random_pet.virtual_machine_names (expand)" references: []
2024-02-01T20:31:16.332-0500 [DEBUG] ReferenceTransformer: "module.virtual_machines.var.domain (expand)" references: []
2024-02-01T20:31:16.332-0500 [DEBUG] ReferenceTransformer: "module.download_files (expand)" references: []
2024-02-01T20:31:16.332-0500 [DEBUG] ReferenceTransformer: "module.virtual_machines.var.virtual_machines (expand)" references: []
2024-02-01T20:31:16.332-0500 [DEBUG] ReferenceTransformer: "module.download_files.var.node (expand)" references: []
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [INFO]  ReferenceTransformer: reference not found: "each.value"
2024-02-01T20:31:16.333-0500 [DEBUG] ReferenceTransformer: "module.virtual_machines.proxmox_virtual_environment_vm.virtual_machines[\"0\"]" references: [module.virtual_machines.var.virtual_machines (expand) module.virtual_machines.var.domain (expand) module.virtual_machines.random_pet.virtual_machine_names (expand) module.virtual_machines.var.domain (expand) module.virtual_machines.var.node (expand) module.virtual_machines.var.datastore (expand) module.virtual_machines.var.datastore (expand) module.virtual_machines.var.datastore (expand)]
2024-02-01T20:31:16.333-0500 [DEBUG] ReferenceTransformer: "module.download_files.proxmox_virtual_environment_download_file.download_files (expand)" references: [module.download_files.var.files (expand)]
2024-02-01T20:31:16.333-0500 [DEBUG] ReferenceTransformer: "module.download_files.var.datastore (expand)" references: []
2024-02-01T20:31:16.333-0500 [DEBUG] ReferenceTransformer: "module.virtual_machines.var.node (expand)" references: []
2024-02-01T20:31:16.333-0500 [DEBUG] ReferenceTransformer: "module.virtual_machines.var.datastore (expand)" references: []
2024-02-01T20:31:16.333-0500 [DEBUG] ReferenceTransformer: "module.virtual_machines (expand)" references: [module.download_files (close)]
2024-02-01T20:31:16.333-0500 [DEBUG] ReferenceTransformer: "module.virtual_machines.proxmox_virtual_environment_vm.virtual_machines (expand)" references: [module.virtual_machines.var.virtual_machines (expand)]
2024-02-01T20:31:16.335-0500 [DEBUG] pruneUnusedNodes: module.download_files.var.node (expand) is no longer needed, removing
2024-02-01T20:31:16.335-0500 [DEBUG] pruneUnusedNodes: module.download_files.proxmox_virtual_environment_download_file.download_files (expand) is no longer needed, removing
2024-02-01T20:31:16.335-0500 [DEBUG] pruneUnusedNodes: module.download_files.var.datastore (expand) is no longer needed, removing
2024-02-01T20:31:16.335-0500 [DEBUG] pruneUnusedNodes: module.download_files.var.files (expand) is no longer needed, removing
2024-02-01T20:31:16.335-0500 [DEBUG] pruneUnusedNodes: module.download_files (expand) is no longer needed, removing
2024-02-01T20:31:16.339-0500 [DEBUG] Starting graph walk: walkApply
2024-02-01T20:31:16.340-0500 [DEBUG] created provider logger: level=debug
2024-02-01T20:31:16.340-0500 [INFO]  provider: configuring client automatic mTLS
2024-02-01T20:31:16.353-0500 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/hashicorp/random/3.6.0/darwin_amd64/terraform-provider-random_v3.6.0_x5 args=[.terraform/providers/registry.terraform.io/hashicorp/random/3.6.0/darwin_amd64/terraform-provider-random_v3.6.0_x5]
2024-02-01T20:31:16.363-0500 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/hashicorp/random/3.6.0/darwin_amd64/terraform-provider-random_v3.6.0_x5 pid=57883
2024-02-01T20:31:16.363-0500 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/hashicorp/random/3.6.0/darwin_amd64/terraform-provider-random_v3.6.0_x5
2024-02-01T20:31:16.386-0500 [INFO]  provider.terraform-provider-random_v3.6.0_x5: configuring server automatic mTLS: timestamp=2024-02-01T20:31:16.386-0500
2024-02-01T20:31:16.408-0500 [DEBUG] provider.terraform-provider-random_v3.6.0_x5: plugin address: network=unix address=/var/folders/dg/87sbp66j1y70xf2s4pgf00rc0000gn/T/plugin43858332 timestamp=2024-02-01T20:31:16.408-0500
2024-02-01T20:31:16.408-0500 [DEBUG] provider: using plugin: version=5
2024-02-01T20:31:16.422-0500 [DEBUG] created provider logger: level=debug
2024-02-01T20:31:16.422-0500 [INFO]  provider: configuring client automatic mTLS
2024-02-01T20:31:16.427-0500 [DEBUG] No provider meta schema returned
2024-02-01T20:31:16.429-0500 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/bpg/proxmox/0.46.1/darwin_amd64/terraform-provider-proxmox_v0.46.1 args=[.terraform/providers/registry.terraform.io/bpg/proxmox/0.46.1/darwin_amd64/terraform-provider-proxmox_v0.46.1]
2024-02-01T20:31:16.437-0500 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/bpg/proxmox/0.46.1/darwin_amd64/terraform-provider-proxmox_v0.46.1 pid=57884
2024-02-01T20:31:16.437-0500 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/bpg/proxmox/0.46.1/darwin_amd64/terraform-provider-proxmox_v0.46.1
2024-02-01T20:31:16.458-0500 [INFO]  provider.terraform-provider-proxmox_v0.46.1: configuring server automatic mTLS: timestamp=2024-02-01T20:31:16.458-0500
2024-02-01T20:31:16.471-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1: plugin address: address=/var/folders/dg/87sbp66j1y70xf2s4pgf00rc0000gn/T/plugin1717751949 network=unix timestamp=2024-02-01T20:31:16.471-0500
2024-02-01T20:31:16.471-0500 [DEBUG] provider: using plugin: version=6
2024-02-01T20:31:16.488-0500 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-02-01T20:31:16.490-0500 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/random/3.6.0/darwin_amd64/terraform-provider-random_v3.6.0_x5 pid=57883
2024-02-01T20:31:16.490-0500 [DEBUG] provider: plugin exited
2024-02-01T20:31:16.502-0500 [INFO]  provider.terraform-provider-proxmox_v0.46.1: Configuring the Proxmox provider...: @module=proxmox tf_mux_provider=*proto6server.Server tf_req_id=3d007678-ac09-c088-1102-626f3ce1f87f tf_rpc=ConfigureProvider @caller=github.com/bpg/terraform-provider-proxmox/fwprovider/provider.go:221 tf_provider_addr=registry.terraform.io/bpg/proxmox timestamp=2024-02-01T20:31:16.502-0500
2024-02-01T20:31:16.518-0500 [WARN]  provider.terraform-provider-proxmox_v0.46.1: unable to require attribute replacement: @module=sdk.helper_schema tf_mux_provider=tf5to6server.v5tov6Server tf_provider_addr=registry.terraform.io/bpg/proxmox tf_rpc=PlanResourceChange @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/customdiff/force_new.go:32 error="ForceNew: No changes for vm_id" tf_attribute_path=vm_id tf_req_id=2993bc3e-ceb3-8348-ed88-5ca306a458ad tf_resource_type=proxmox_virtual_environment_vm timestamp=2024-02-01T20:31:16.518-0500
2024-02-01T20:31:16.521-0500 [WARN]  Provider "registry.terraform.io/bpg/proxmox" produced an invalid plan for module.virtual_machines.proxmox_virtual_environment_vm.virtual_machines["0"], but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .migrate: planned value cty.False for a non-computed attribute
      - .started: planned value cty.True for a non-computed attribute
      - .tablet_device: planned value cty.True for a non-computed attribute
      - .scsi_hardware: planned value cty.StringVal("virtio-scsi-pci") for a non-computed attribute
      - .stop_on_destroy: planned value cty.False for a non-computed attribute
      - .timeout_create: planned value cty.NumberIntVal(1800) for a non-computed attribute
      - .timeout_shutdown_vm: planned value cty.NumberIntVal(1800) for a non-computed attribute
      - .timeout_stop_vm: planned value cty.NumberIntVal(300) for a non-computed attribute
      - .timeout_clone: planned value cty.NumberIntVal(1800) for a non-computed attribute
      - .timeout_migrate: planned value cty.NumberIntVal(1800) for a non-computed attribute
      - .keyboard_layout: planned value cty.StringVal("en-us") for a non-computed attribute
      - .reboot: planned value cty.False for a non-computed attribute
      - .timeout_move_disk: planned value cty.NumberIntVal(1800) for a non-computed attribute
      - .timeout_reboot: planned value cty.NumberIntVal(1800) for a non-computed attribute
      - .on_boot: planned value cty.True for a non-computed attribute
      - .template: planned value cty.False for a non-computed attribute
      - .timeout_start_vm: planned value cty.NumberIntVal(1800) for a non-computed attribute
      - .acpi: planned value cty.True for a non-computed attribute
      - .memory[0].floating: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .memory[0].shared: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .disk[0].cache: planned value cty.StringVal("none") for a non-computed attribute
      - .cpu[0].units: planned value cty.NumberIntVal(1024) for a non-computed attribute
      - .cpu[0].architecture: planned value cty.StringVal("x86_64") for a non-computed attribute
      - .cpu[0].hotplugged: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .cpu[0].limit: planned value cty.NumberIntVal(0) for a non-computed attribute
      - .cpu[0].numa: planned value cty.False for a non-computed attribute
      - .cpu[0].sockets: planned value cty.NumberIntVal(1) for a non-computed attribute
      - .initialization[0].datastore_id: planned value cty.StringVal("local-lvm") for a non-computed attribute
module.virtual_machines.proxmox_virtual_environment_vm.virtual_machines["0"]: Creating...
2024-02-01T20:31:16.522-0500 [INFO]  Starting apply for module.virtual_machines.proxmox_virtual_environment_vm.virtual_machines["0"]
2024-02-01T20:31:16.523-0500 [DEBUG] module.virtual_machines.proxmox_virtual_environment_vm.virtual_machines["0"]: applying the planned Create change
2024-02-01T20:31:16.525-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1: 2024/02/01 20:31:16 [DEBUG] setting computed for "network_interface_names" from ComputedKeys
2024-02-01T20:31:16.525-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1: 2024/02/01 20:31:16 [DEBUG] setting computed for "ipv4_addresses" from ComputedKeys
2024-02-01T20:31:16.525-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1: 2024/02/01 20:31:16 [DEBUG] setting computed for "ipv6_addresses" from ComputedKeys
2024-02-01T20:31:16.525-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1: 2024/02/01 20:31:16 [DEBUG] setting computed for "mac_addresses" from ComputedKeys
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1: panic: interface conversion: interface {} is nil, not map[string]interface {}
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1: 
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1: goroutine 177 [running]:
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1: github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource.vmGetCloudInitConfig(0x19e1140?)
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1:       github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm.go:3185 +0x103b
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1: github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource.vmCreateCustom({0x1c96bd0, 0xc0000e1030}, 0x22cfe40?, {0x1a5aba0?, 0xc000326d80?})
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1:       github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm.go:2602 +0x12d3
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1: github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource.vmCreate({0x1c96bd0, 0xc0000e1030}, 0x0?, {0x1a5aba0, 0xc000326d80})
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1:       github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm.go:1688 +0x94
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc00030d340, {0x1c96b28, 0xc000495140}, 0xd?, {0x1a5aba0, 0xc000326d80})
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1:       github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:778 +0x11b
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc00030d340, {0x1c96b28, 0xc000495140}, 0xc00028c340, 0xc000548e80, {0x1a5aba0, 0xc000326d80})
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1:       github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:909 +0xa89
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000011ad0, {0x1c96b28?, 0xc000494f90?}, 0xc0004a0e10)
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1:       github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/grpc_provider.go:1074 +0xdbc
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1: github.com/hashicorp/terraform-plugin-mux/tf5to6server.v5tov6Server.ApplyResourceChange({{0x1c9e340?, 0xc000011ad0?}}, {0x1c96b28, 0xc000494f90}, 0x0?)
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1:       github.com/hashicorp/terraform-plugin-mux@v0.13.0/tf5to6server/tf5to6server.go:37 +0x54
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1: github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ApplyResourceChange(0x1c96b60?, {0x1c96b28?, 0xc000494c90?}, 0xc0004a0dc0)
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1:       github.com/hashicorp/terraform-plugin-mux@v0.13.0/tf6muxserver/mux_server_ApplyResourceChange.go:36 +0x193
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1: github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0xc0002739a0, {0x1c96b28?, 0xc000494270?}, 0xc0000e0690)
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1:       github.com/hashicorp/terraform-plugin-go@v0.21.0/tfprotov6/tf6server/server.go:845 +0x3d0
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1: github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x1adffc0?, 0xc0002739a0}, {0x1c96b28, 0xc000494270}, 0xc000548000, 0x0)
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1:       github.com/hashicorp/terraform-plugin-go@v0.21.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:518 +0x169
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1: google.golang.org/grpc.(*Server).processUnaryRPC(0xc00019d000, {0x1c96b28, 0xc0004941e0}, {0x1c9c8e8, 0xc000480680}, 0xc00058a000, 0xc0003be0f0, 0x22c30b8, 0x0)
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1:       google.golang.org/grpc@v1.61.0/server.go:1385 +0xe03
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1: google.golang.org/grpc.(*Server).handleStream(0xc00019d000, {0x1c9c8e8, 0xc000480680}, 0xc00058a000)
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1:       google.golang.org/grpc@v1.61.0/server.go:1796 +0xfec
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1: google.golang.org/grpc.(*Server).serveStreams.func2.1()
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1:       google.golang.org/grpc@v1.61.0/server.go:1029 +0x8b
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1: created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 23
2024-02-01T20:31:16.531-0500 [DEBUG] provider.terraform-provider-proxmox_v0.46.1:       google.golang.org/grpc@v1.61.0/server.go:1040 +0x135
2024-02-01T20:31:16.533-0500 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/bpg/proxmox/0.46.1/darwin_amd64/terraform-provider-proxmox_v0.46.1 pid=57884 error="exit status 2"
2024-02-01T20:31:16.533-0500 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-02-01T20:31:16.534-0500 [ERROR] plugin6.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-02-01T20:31:16.543-0500 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2024-02-01T20:31:16.543-0500 [ERROR] vertex "module.virtual_machines.proxmox_virtual_environment_vm.virtual_machines[\"0\"]" error: Plugin did not respond
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-proxmox_v0.46.1 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 177 [running]:
github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource.vmGetCloudInitConfig(0x19e1140?)
        github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm.go:3185 +0x103b
github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource.vmCreateCustom({0x1c96bd0, 0xc0000e1030}, 0x22cfe40?, {0x1a5aba0?, 0xc000326d80?})
        github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm.go:2602 +0x12d3
github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource.vmCreate({0x1c96bd0, 0xc0000e1030}, 0x0?, {0x1a5aba0, 0xc000326d80})
        github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm.go:1688 +0x94
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc00030d340, {0x1c96b28, 0xc000495140}, 0xd?, {0x1a5aba0, 0xc000326d80})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:778 +0x11b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc00030d340, {0x1c96b28, 0xc000495140}, 0xc00028c340, 0xc000548e80, {0x1a5aba0, 0xc000326d80})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:909 +0xa89
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000011ad0, {0x1c96b28?, 0xc000494f90?}, 0xc0004a0e10)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/grpc_provider.go:1074 +0xdbc
github.com/hashicorp/terraform-plugin-mux/tf5to6server.v5tov6Server.ApplyResourceChange({{0x1c9e340?, 0xc000011ad0?}}, {0x1c96b28, 0xc000494f90}, 0x0?)
        github.com/hashicorp/terraform-plugin-mux@v0.13.0/tf5to6server/tf5to6server.go:37 +0x54
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ApplyResourceChange(0x1c96b60?, {0x1c96b28?, 0xc000494c90?}, 0xc0004a0dc0)
        github.com/hashicorp/terraform-plugin-mux@v0.13.0/tf6muxserver/mux_server_ApplyResourceChange.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0xc0002739a0, {0x1c96b28?, 0xc000494270?}, 0xc0000e0690)
        github.com/hashicorp/terraform-plugin-go@v0.21.0/tfprotov6/tf6server/server.go:845 +0x3d0
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x1adffc0?, 0xc0002739a0}, {0x1c96b28, 0xc000494270}, 0xc000548000, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.21.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:518 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00019d000, {0x1c96b28, 0xc0004941e0}, {0x1c9c8e8, 0xc000480680}, 0xc00058a000, 0xc0003be0f0, 0x22c30b8, 0x0)
        google.golang.org/grpc@v1.61.0/server.go:1385 +0xe03
google.golang.org/grpc.(*Server).handleStream(0xc00019d000, {0x1c9c8e8, 0xc000480680}, 0xc00058a000)
        google.golang.org/grpc@v1.61.0/server.go:1796 +0xfec
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.61.0/server.go:1029 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 23
        google.golang.org/grpc@v1.61.0/server.go:1040 +0x135

Error: The terraform-provider-proxmox_v0.46.1 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

2024-02-01T20:31:16.550-0500 [DEBUG] provider: plugin exited
```

--- a/proxmoxtf/resource/vm.go
+++ b/proxmoxtf/resource/vm.go
@@ -1997,7 +1997,7 @@ func vmCreateClone(ctx context.Context, d *schema.ResourceData, m interface{}) d
 		updateBody.ACPI = &acpi
 	}
 
-	if len(agent) > 0 {
+	if len(agent) > 0 && agent[0] != nil {
 		agentBlock := agent[0].(map[string]interface{})
 
 		agentEnabled := types.CustomBool(
@@ -2042,7 +2042,7 @@ func vmCreateClone(ctx context.Context, d *schema.ResourceData, m interface{}) d
 		}
 	}
 
-	if len(cdrom) > 0 {
+	if len(cdrom) > 0 && cdrom[0] != nil {
 		cdromBlock := cdrom[0].(map[string]interface{})
 
 		cdromEnabled := cdromBlock[mkResourceVirtualEnvironmentVMCDROMEnabled].(bool)
@@ -2062,7 +2062,7 @@ func vmCreateClone(ctx context.Context, d *schema.ResourceData, m interface{}) d
 		}
 	}
 
-	if len(cpu) > 0 {
+	if len(cpu) > 0 && cpu[0] != nil {
 		cpuBlock := cpu[0].(map[string]interface{})
 
 		cpuArchitecture := cpuBlock[mkResourceVirtualEnvironmentVMCPUArchitecture].(string)
@@ -2110,7 +2110,7 @@ func vmCreateClone(ctx context.Context, d *schema.ResourceData, m interface{}) d
 		return diag.FromErr(err)
 	}
 
-	if len(initialization) > 0 {
+	if len(initialization) > 0 && initialization[0] != nil {
 		tflog.Trace(ctx, "Preparing the CloudInit configuration")
 
 		initializationBlock := initialization[0].(map[string]interface{})
@@ -2157,7 +2157,7 @@ func vmCreateClone(ctx context.Context, d *schema.ResourceData, m interface{}) d
 		updateBody.KeyboardLayout = &keyboardLayout
 	}
 
-	if len(memory) > 0 {
+	if len(memory) > 0 && memory[0] != nil {
 		memoryBlock := memory[0].(map[string]interface{})
 
 		memoryDedicated := memoryBlock[mkResourceVirtualEnvironmentVMMemoryDedicated].(int)
@@ -2191,7 +2191,7 @@ func vmCreateClone(ctx context.Context, d *schema.ResourceData, m interface{}) d
 		}
 	}
 
-	if len(operatingSystem) > 0 {
+	if len(operatingSystem) > 0 && operatingSystem[0] != nil {
 		operatingSystemBlock := operatingSystem[0].(map[string]interface{})
 		operatingSystemType := operatingSystemBlock[mkResourceVirtualEnvironmentVMOperatingSystemType].(string)
 
@@ -2567,7 +2567,7 @@ func vmCreateCustom(ctx context.Context, d *schema.ResourceData, m interface{}) 
 	var efiDisk *vms.CustomEFIDisk
 
 	efiDiskBlock := d.Get(mkResourceVirtualEnvironmentVMEFIDisk).([]interface{})
-	if len(efiDiskBlock) > 0 {
+	if len(efiDiskBlock) > 0 && efiDiskBlock[0] != nil {
 		block := efiDiskBlock[0].(map[string]interface{})
 
 		datastoreID, _ := block[mkResourceVirtualEnvironmentVMEFIDiskDatastoreID].(string)
@@ -2590,7 +2590,7 @@ func vmCreateCustom(ctx context.Context, d *schema.ResourceData, m interface{}) 
 	var tpmState *vms.CustomTPMState
 
 	tpmStateBlock := d.Get(mkResourceVirtualEnvironmentVMTPMState).([]interface{})
-	if len(tpmStateBlock) > 0 {
+	if len(tpmStateBlock) > 0 && tpmStateBlock[0] != nil {
 		block := tpmStateBlock[0].(map[string]interface{})
 
 		datastoreID, _ := block[mkResourceVirtualEnvironmentVMTPMStateDatastoreID].(string)
@@ -3148,16 +3148,13 @@ func vmGetCloudInitConfig(d *schema.ResourceData) *vms.CustomCloudInitConfig {
 	}
 
 	initializationIPConfig := initializationBlock[mkResourceVirtualEnvironmentVMInitializationIPConfig].([]interface{})
-	initializationConfig.IPConfig = make(
-		[]vms.CustomCloudInitIPConfig,
-		len(initializationIPConfig),
-	)
+	initializationConfig.IPConfig = make([]vms.CustomCloudInitIPConfig, len(initializationIPConfig))
 
 	for i, c := range initializationIPConfig {
 		configBlock := c.(map[string]interface{})
 		ipv4 := configBlock[mkResourceVirtualEnvironmentVMInitializationIPConfigIPv4].([]interface{})
 
-		if len(ipv4) > 0 {
+		if len(ipv4) > 0 && ipv4[0] != nil {
 			ipv4Block := ipv4[0].(map[string]interface{})
 			ipv4Address := ipv4Block[mkResourceVirtualEnvironmentVMInitializationIPConfigIPv4Address].(string)
 
@@ -3174,7 +3171,7 @@ func vmGetCloudInitConfig(d *schema.ResourceData) *vms.CustomCloudInitConfig {
 
 		ipv6 := configBlock[mkResourceVirtualEnvironmentVMInitializationIPConfigIPv6].([]interface{})
 
-		if len(ipv6) > 0 {
+		if len(ipv6) > 0 && ipv6[0] != nil {
 			ipv6Block := ipv6[0].(map[string]interface{})
 			ipv6Address := ipv6Block[mkResourceVirtualEnvironmentVMInitializationIPConfigIPv6Address].(string)
 
@@ -3192,7 +3189,7 @@ func vmGetCloudInitConfig(d *schema.ResourceData) *vms.CustomCloudInitConfig {
 
 	initializationUserAccount := initializationBlock[mkResourceVirtualEnvironmentVMInitializationUserAccount].([]interface{})
 
-	if len(initializationUserAccount) > 0 {
+	if len(initializationUserAccount) > 0 && initializationUserAccount[0] != nil {
 		initializationUserAccountBlock := initializationUserAccount[0].(map[string]interface{})
 		keys := initializationUserAccountBlock[mkResourceVirtualEnvironmentVMInitializationUserAccountKeys].([]interface{})
 
@@ -3407,7 +3404,7 @@ func vmGetEfiDisk(d *schema.ResourceData, disk []interface{}) *vms.CustomEFIDisk
 
 	var efiDiskConfig *vms.CustomEFIDisk
 
-	if len(efiDisk) > 0 {
+	if len(efiDisk) > 0 && efiDisk[0] != nil {
 		efiDiskConfig = &vms.CustomEFIDisk{}
 
 		block := efiDisk[0].(map[string]interface{})
@@ -3469,7 +3466,7 @@ func vmGetTPMState(d *schema.ResourceData, disk []interface{}) *vms.CustomTPMSta
 
 	var tpmStateConfig *vms.CustomTPMState
 
-	if len(tpmState) > 0 {
+	if len(tpmState) > 0 && tpmState[0] != nil {
 		tpmStateConfig = &vms.CustomTPMState{}
 
 		block := tpmState[0].(map[string]interface{})
@@ -3665,7 +3662,7 @@ func vmGetSerialDeviceList(d *schema.ResourceData) vms.CustomSerialDevices {
 func vmGetSMBIOS(d *schema.ResourceData) *vms.CustomSMBIOS {
 	smbiosSections := d.Get(mkResourceVirtualEnvironmentVMSMBIOS).([]interface{})
 	//nolint:nestif
-	if len(smbiosSections) > 0 {
+	if len(smbiosSections) > 0 && smbiosSections[0] != nil {
 		smbiosBlock := smbiosSections[0].(map[string]interface{})
 		b64 := types.CustomBool(true)
 		family, _ := smbiosBlock[mkResourceVirtualEnvironmentVMSMBIOSFamily].(string)
@@ -3726,7 +3723,8 @@ func vmGetSMBIOS(d *schema.ResourceData) *vms.CustomSMBIOS {
 
 func vmGetStartupOrder(d *schema.ResourceData) *vms.CustomStartupOrder {
 	startup := d.Get(mkResourceVirtualEnvironmentVMStartup).([]interface{})
-	if len(startup) > 0 {
+
+	if len(startup) > 0 && startup[0] != nil {
 		startupBlock := startup[0].(map[string]interface{})
 		startupOrder := startupBlock[mkResourceVirtualEnvironmentVMStartupOrder].(int)
 		startupUpDelay := startupBlock[mkResourceVirtualEnvironmentVMStartupUpDelay].(int)
@@ -3957,7 +3955,7 @@ func vmReadCustom(
 				agent[mkResourceVirtualEnvironmentVMAgentTrim] = false
 			}
 
-			if len(currentAgent) > 0 {
+			if len(currentAgent) > 0 && currentAgent[0] != nil {
 				currentAgentBlock := currentAgent[0].(map[string]interface{})
 				currentAgentTimeout := currentAgentBlock[mkResourceVirtualEnvironmentVMAgentTimeout].(string)
 
@@ -4042,7 +4040,7 @@ func vmReadCustom(
 	currentInterface := dvResourceVirtualEnvironmentVMCDROMInterface
 
 	currentCDROM := d.Get(mkResourceVirtualEnvironmentVMCDROM).([]interface{})
-	if len(currentCDROM) > 0 {
+	if len(currentCDROM) > 0 && currentCDROM[0] != nil {
 		currentBlock := currentCDROM[0].(map[string]interface{})
 		currentInterface = currentBlock[mkResourceVirtualEnvironmentVMCDROMInterface].(string)
 	}
@@ -4059,7 +4057,7 @@ func vmReadCustom(
 			cdromBlock[mkResourceVirtualEnvironmentVMCDROMFileID] = cdromIDEDevice.FileVolume
 			cdromBlock[mkResourceVirtualEnvironmentVMCDROMInterface] = currentInterface
 
-			if len(currentCDROM) > 0 {
+			if len(currentCDROM) > 0 && currentCDROM[0] != nil {
 				currentBlock := currentCDROM[0].(map[string]interface{})
 
 				if currentBlock[mkResourceVirtualEnvironmentVMCDROMFileID] == "" {
@@ -4498,11 +4496,11 @@ func vmReadCustom(
 		currentInitializationDNSBlock := map[string]interface{}{}
 		currentInitialization := d.Get(mkResourceVirtualEnvironmentVMInitialization).([]interface{})
 
-		if len(currentInitialization) > 0 {
+		if len(currentInitialization) > 0 && currentInitialization[0] != nil {
 			currentInitializationBlock := currentInitialization[0].(map[string]interface{})
 			//nolint:lll
 			currentInitializationDNS := currentInitializationBlock[mkResourceVirtualEnvironmentVMInitializationDNS].([]interface{})
-			if len(currentInitializationDNS) > 0 {
+			if len(currentInitializationDNS) > 0 && currentInitializationDNS[0] != nil {
 				currentInitializationDNSBlock = currentInitializationDNS[0].(map[string]interface{})
 			}
 		}

